frontend/src/metabase/parameters/components/ParameterSettings/TemporalUnitSettings/TemporalUnitSettings.unit.spec.tsx

```diff
@@ -0,0 +1,118 @@
+import userEvent from "@testing-library/user-event";
+
+import { render, screen } from "__support__/ui";
+import * as Lib from "metabase-lib";
+import type { Parameter, TemporalUnit } from "metabase-types/api";
+import { createMockParameter } from "metabase-types/api/mocks";
+
+import { TemporalUnitSettings } from "./TemporalUnitSettings";
+
+interface SetupOpts {
+  parameter?: Parameter;
+}
+
+function setup({ parameter = createMockParameter() }: SetupOpts = {}) {
+  const onChangeTemporalUnits = jest.fn();
+
+  render(
+    <TemporalUnitSettings
+      parameter={parameter}
+      onChangeTemporalUnits={onChangeTemporalUnits}
+    />,
+  );
+
+  return { onChangeTemporalUnits };
+}
+
+describe("TemporalUnitSettings", () => {
+  it.each<[string, TemporalUnit[] | undefined]>([
+    ["All", undefined],
+    ["All", Lib.availableTemporalUnits()],
+    ["None", []],
+    ["Minute", ["minute"]],
+    ["Month, Quarter", ["month", "quarter"]],
+    ["Month, Quarter, Day of week", ["month", "quarter", "day-of-week"]],
+    ["Day, Month, Quarter, +1", ["day", "month", "quarter", "year"]],
+    ["Hour, Day, Month, +2", ["hour", "day", "month", "quarter", "year"]],
+  ])("should correctly format %s", async (value, units) => {
+    setup({
+      parameter: createMockParameter({
+        type: "temporal-unit",
+        temporal_units: units,
+      }),
+    });
+
+    expect(await screen.findByText(value)).toBeInTheDocument();
+  });
+
+  it("should select a unit and preserve the unit order", async () => {
+    const { onChangeTemporalUnits } = setup({
+      parameter: createMockParameter({
+        type: "temporal-unit",
+        temporal_units: ["day", "year"],
+      }),
+    });
+    await userEvent.click(await screen.findByText("Day, Year"));
+    await userEvent.click(await screen.findByLabelText("Month"));
+    expect(onChangeTemporalUnits).toHaveBeenCalledWith([
+      "day",
+      "month",
+      "year",
+    ]);
+  });
+
+  it("should deselect a unit and preserve the unit order", async () => {
```

Comment:
Preserving the order in the callback is a bit unusually strict - I wouldn't worry about it and instead did something like this in `TemporalUnitWidget`:

```tsx
const temporalUnits = Lib.availableTemporalUnits().filter((unit) => parameter.temporal_unit.includes(unit));
```

But it's fine like this too.